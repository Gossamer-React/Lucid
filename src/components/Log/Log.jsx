import React from "react";

const Log = props => {
  console.log("---LOG---", props);

  return (
    <div className="log">
      <p className="log-p">
        <b>Request:</b> {props.logId}
      </p>
      <p className="log-p">
        <b>Operation Name:</b> {props.operationName}
      </p>
      <p className="log-p">
        <b>Query:</b> {props.query}
      </p>
      <p className="log-p">
        <b>Variable:</b> {JSON.stringify(props.variables)}
      </p>
    </div>
  );
};

export default Log;
