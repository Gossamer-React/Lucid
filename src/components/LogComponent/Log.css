.log{
  background-color: rgba(255, 255, 255);
  box-sizing: border-box;
  box-shadow: 0 15px 30px 0 rgba(0,0,0,0.11),
  0 5px 15px 0 rgba(0,0,0,0.08);
  /* transition: box-shadow .1s; */
  cursor: pointer;
  margin: 15px;
  overflow-x: scroll;
  padding: 15px;
}

.log-p{
  margin: 5px 0px;
}

.log-span{
  font-size: 1.2em;
  font-weight: 500;
  letter-spacing: 0.50px;
  margin: 5px 0px;
}