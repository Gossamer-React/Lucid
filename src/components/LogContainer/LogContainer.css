/* *for log in the graphql tab */
#log-container{
  grid-area: log-container;
  height: 500px;
  margin-bottom: 30px;
} 

#log-header{
  display: flex;
  justify-content: space-between;
  margin-bottom: 5px;
}

#logs{
  overflow-y: scroll;
  max-height: 95%;
}

/* *for responsive use */
@media screen and (min-width: 760px) {
  #log-container{
    width: 100%;
    margin: 0 auto 30px auto;
  } 
}