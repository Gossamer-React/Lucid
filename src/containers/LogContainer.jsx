import React, { Component } from 'react';
import Log from './../components/Log/Log.jsx';
import styles from '../../public/log-container.css';

class LogContainer extends Component {
  constructor(props) { 
    super(props);
    this.state = {
      logs: _Logs
    }
  }

  render() {
    console.log("LOGS:", _Logs);
    let logs = this.state.logs.map((log) => {
      console.log(log);
      return  <Log />;
    });

    return (
      <div id="log-container">
        <h1>Log</h1>
        {logs}
      </div>
    )
  }
}

export default LogContainer;
