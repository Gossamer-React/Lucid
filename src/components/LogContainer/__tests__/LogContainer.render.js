import React from 'react';
import renderer from 'react-test-renderer';
import LogContainer from './../LogContainer';
import Log from './../../LogComponent/Log.jsx';

//* Tests to make sure that comopnent does not change unexpectdly.
describe('<LogContainer />', () => {
  it('Will render the LogContainer with an emapy array of logs and match the snapshot', ()=> {
    const logContainer = renderer.create(
      <LogContainer
        logs={[]}
      />
    ).toJSON();

    expect(logContainer).toMatchSnapshot();
  });

  it('Will render the LogContainer with one log and match the snapshot', ()=> {
    const logContainer = renderer.create(
      <LogContainer
        logs={[{
          req: {
            postData: {
              text:JSON.stringify({
                operationName: 'Request One',
                query: 'This is the query.',
                variables: { title: 'laundry' }
              })
            }
          }
        }]}
      />
    ).toJSON();

    expect(logContainer).toMatchSnapshot();
  });
});
