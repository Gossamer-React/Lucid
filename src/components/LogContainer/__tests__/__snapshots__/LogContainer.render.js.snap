// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogContainer /> Will render the LogContainer with an emapy array of logs and match the snapshot 1`] = `
<h2>
  No requests have been made yet.
</h2>
`;

exports[`<LogContainer /> Will render the LogContainer with one log and match the snapshot 1`] = `
<div
  id="log-container"
>
  <div
    id="log-header"
  >
    <h2>
      Request Log
    </h2>
    <button
      className="appButton"
      data-log="req-log"
      onClick={[Function]}
    >
      Clear
    </button>
  </div>
  <div
    id="logs"
  >
    <div
      className="log"
      onClick={[Function]}
    >
      <p
        className="log-p"
      >
        <b>
          Request:
        </b>
         
        0
      </p>
      <p
        className="log-p"
      >
        <b>
          Operation Name:
        </b>
         
        Request One
      </p>
      <p
        className="log-p"
      >
        <b>
          Query:
        </b>
        <div>
          [PARSE ERROR] 
          This is the query.
        </div>
      </p>
      <p
        className="log-p"
      >
        <b>
          Variables:
        </b>
        <span
          className="log-span"
        >
          <div
            className="react-json-view"
            style={
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0)",
                "cursor": "default",
                "fontFamily": "monospace",
                "position": "relative",
              }
            }
          >
            <div
              className="pretty-json-container object-container"
            >
              <div
                className="object-content"
              >
                <div
                  className="object-key-val"
                >
                  <span>
                    <span
                      onClick={[Function]}
                      style={
                        Object {
                          "cursor": "pointer",
                          "display": "inline-block",
                        }
                      }
                    >
                      <div
                        className="icon-container"
                        style={
                          Object {
                            "display": "inline-block",
                            "width": "17px",
                          }
                        }
                      >
                        <span
                          className="expanded-icon"
                        >
                          <svg
                            fill="currentColor"
                            style={
                              Object {
                                "color": "#586e75",
                                "height": "1em",
                                "paddingLeft": "2px",
                                "verticalAlign": "top",
                                "width": "1em",
                              }
                            }
                            viewBox="0 0 15 15"
                          >
                            <path
                              d="M0 5l6 6 6-6z"
                            />
                          </svg>
                        </span>
                      </div>
                      <span />
                      <span
                        style={
                          Object {
                            "color": "#002b36",
                            "cursor": "pointer",
                            "display": "inline-block",
                            "fontWeight": "bold",
                          }
                        }
                      >
                        {
                      </span>
                    </span>
                    <div
                      className="object-meta-data"
                      onClick={[Function]}
                      style={
                        Object {
                          "display": "inline-block",
                          "padding": "0px 0px 0px 10px",
                        }
                      }
                    >
                      <span
                        className="object-size"
                        style={
                          Object {
                            "borderRadius": "3px",
                            "color": "rgba(0, 0, 0, 0.3)",
                            "cursor": "default",
                            "fontStyle": "italic",
                            "margin": "0px 6px 0px 0px",
                          }
                        }
                      >
                        1
                         item
                        
                      </span>
                    </div>
                  </span>
                  <div
                    className="pushed-content object-container"
                  >
                    <div
                      className="object-content"
                      style={
                        Object {
                          "marginLeft": "6px",
                        }
                      }
                    >
                      <div
                        className="variable-row"
                        style={
                          Object {
                            ":hover": Object {
                              "borderLeft": "2px solid rgb(235, 235, 235)",
                              "paddingLeft": "4px",
                            },
                            "borderLeft": "1px solid rgb(235, 235, 235)",
                            "paddingBottom": "3px",
                            "paddingLeft": 5,
                            "paddingRight": "5px",
                            "paddingTop": "3px",
                          }
                        }
                      >
                        <span>
                          <span
                            className="object-key"
                            style={
                              Object {
                                ":hover": Object {
                                  "opacity": "1",
                                },
                                "color": "#002b36",
                                "display": "inline-block",
                                "fontStyle": "none",
                                "letterSpacing": "0.5px",
                                "opacity": "0.85",
                                "verticalAlign": "top",
                              }
                            }
                          >
                            <span
                              style={
                                Object {
                                  "verticalAlign": "top",
                                }
                              }
                            >
                              "
                            </span>
                            <span
                              style={
                                Object {
                                  "display": "inline-block",
                                }
                              }
                            >
                              title
                            </span>
                            <span
                              style={
                                Object {
                                  "verticalAlign": "top",
                                }
                              }
                            >
                              "
                            </span>
                          </span>
                          <span
                            style={
                              Object {
                                "color": "#002b36",
                                "display": "inline-block",
                                "margin": "0px 5px",
                                "verticalAlign": "top",
                              }
                            }
                          >
                            :
                          </span>
                        </span>
                        <div
                          className="variable-value"
                          onClick={null}
                          style={
                            Object {
                              "cursor": "default",
                              "display": "inline-block",
                              "paddingRight": "6px",
                              "position": "relative",
                            }
                          }
                        >
                          <div
                            style={
                              Object {
                                "color": "#cb4b16",
                                "display": "inline-block",
                              }
                            }
                          >
                            <span
                              className="string-value"
                              onClick={[Function]}
                              style={
                                Object {
                                  "cursor": "default",
                                }
                              }
                            >
                              "
                              laundry
                              "
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <span
                    className="brace-row"
                  >
                    <span
                      style={
                        Object {
                          "color": "#002b36",
                          "cursor": "pointer",
                          "display": "inline-block",
                          "fontWeight": "bold",
                          "paddingLeft": "3px",
                        }
                      }
                    >
                      }
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </span>
      </p>
    </div>
  </div>
</div>
`;
